Sources/MongoSwift/ClientSession.swift

```diff
@@ -29,35 +29,10 @@ private func withSessionOpts<T>(
     return try body(opts)
 }
 
-/**
- * A MongoDB client session.
- * This class represents a logical session used for ordering sequential operations.
- *
- * To create a client session, use `startSession` or `withSession` on a `MongoClient` or a `SyncMongoClient`.
- *
- * If `causalConsistency` is not set to `false` when starting a session, read and write operations that use the session
- * will be provided causal consistency guarantees depending on the read and write concerns used. Using "majority"
- * read and write preferences will provide the full set of guarantees. See
- * https://docs.mongodb.com/manual/core/read-isolation-consistency-recency/#sessions for more details.
- *
- * e.g.
- *   ```
- *   let opts = CollectionOptions(readConcern: ReadConcern(.majority), writeConcern: try WriteConcern(w: .majority))
- *   let collection = database.collection("mycoll", options: opts)
- *   try client.withSession { session in
- *       try collection.insertOne(["x": 1], session: session)
- *       try collection.find(["x": 1], session: session)
- *   }
- *   ```
- *
- * To disable causal consistency, set `causalConsistency` to `false` in the `ClientSessionOptions` passed in to either
- * `withSession` or `startSession`.
- *
- * - SeeAlso:
- *   - https://docs.mongodb.com/manual/core/read-isolation-consistency-recency/#sessions
- *   - https://docs.mongodb.com/manual/core/causal-consistency-read-write-concerns/
- */
-public final class SyncClientSession {
+/// A base class for `SyncClientSession` and `AsyncClientSession`.
+public class ClientSession {
+    internal let _client: MongoClient
```

Comment:
we now store the client both in the base class and here. the reason for this is that exposing the client is (for whatever reason) part of the public API for sessions, and it seems odd for a `SyncClientSession` to just return a `MongoClient` rather than a `SyncMongoClient`.
at the same time, the base class needs access to the parent client to implement e.g. `getConnection`.